### ISSUE TITLE
Rule Change: max-statements should count statements in switch cases and braceless bodies

### ISSUE THREAD

**Issue by xbinaryx** [eslint/eslint#21333]
### What rule do you want to change?

max-statements

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint max-statements: ["error", 1] */

function withBraces() {
    if (a) { b(); }
}

function withoutBraces() {
    if (a) b();
}

function withSwitch(x) {
    switch (x) {
        case 1:
            a();
            b();
            c();
            break;
    }
}
```

### What does the rule currently do for this code?

Only `withBraces` is reported. The same two statements without braces, and the statements in `withSwitch`, are not counted at all.

### What will the rule do after it's changed?

Statements are counted whether or not they are wrapped in a block, so all three functions are reported.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by nzakas** [comment]
Yeah, this looks like a bug to fix in semver-minor.

One edge case to be aware of is with empty statements:

```js
if (true);

if (true) {};
```

I think these should return the same count (1).
